packages/fetchai/skills/generic_buyer/tests/test_behaviours.py

```diff
@@ -285,13 +302,13 @@ def test_act_i(self):
         self.assert_quantity_in_outbox(1)
 
         # _start_processing
-        self._check_start_processing_effects(self, fipa_dialogue, mock_logger)
+        self._check_start_processing_effects(self.fipa_dialogue, mock_logger)
 
     def test_act_ii(self):
         """Test the act method of the transaction behaviour where processing is NOT None and processing_time < max_processing."""
         # setup
         processing_time = 5.0
-        self.transaction_behaviour.processing = "some_dialogue"
+        self.transaction_behaviour.processing = "some_dialogue"  # TODO: type
```

Comment:
there are lots of incorrect types used in test of generic_buyer and generic_seller. This is not urgent or important, but fact of the matter is that there are more `[   ERROR]` in stdout than `PASSED`